### ISSUE TITLE
S_CLIENT(1): the TLSv1.2 and TLSv1.3 lists of ciphers are actually not "combined"

### ISSUE THREAD

**Issue by DimitriPapadopoulos** [openssl/openssl#11724]
I'm using OpenSSL 1.1.1. The part of the [S_CLIENT(1)](https://www.openssl.org/docs/man1.1.1/man1/openssl-s_client.html#OPTIONS) that refers to "combining" TLSv1.2 and TLSv1.3 ciphers is confusing:
> **-cipher cipherlist**
> This allows the TLSv1.2 and below cipher list sent by the client to be modified. This list will be combined with any TLSv1.3 ciphersuites that have been configured. Although the server determines which ciphersuite is used it should take the first supported cipher in the list sent by the client. See the **ciphers** command for more information.
> 
> **-ciphersuites val**
 > This allows the TLSv1.3 ciphersuites sent by the client to be modified. This list will be combined with any TLSv1.2 and below ciphersuites that have been configured. Although the server determines which cipher suite is used it should take the first supported cipher in the list sent by the client. See the **ciphers** command for more information. The format for this list is a simple colon (":") separated list of TLSv1.3 ciphersuite names.

It's unclear how the list of TLSv1.2 and TLSv1.3 ciphers are "combined". Actually I would argue they are **not** combined. See for example https://github.com/openssl/openssl/issues/10936#issuecomment-578060278:
> TLSv1.2 (and below) ciphersuites are handled separately to TLSv1.3 ciphersuites. The two are incompatible with each other (you can't use a TLSv1.2 ciphersuite in TLSv1.3 and vice versa).

This is directly related to #7279, although not a duplicate.

**Comment by mattcaswell** [comment]
At a protocol level they *are* combined. When a client sends a ClientHello message to a server is sends a list of all the ciphersuites it supports. This list includes both TLSv1.2 and TLSv1.3 ciphersuites (which could in theory be in any order). In practice though, if TLSv1.3 is selected the server will only choose from the TLSv1.3 ciphersuites, and if TLSv1.2 is selected then it will only choose from the TLSv1.2 ciphersuites.

**Comment by DimitriPapadopoulos** [comment]
Fair enough, at some point and at the protocol level they may be combined. However they are not combined earlier on .For example you cannot combine TLSv1.2 and TLSv1.3 ciphers in **`-cipher`** and **`-ciphersuites`** options:
```
$ openssl s_client -cipher 'ECDHE-RSA-AES256-GCM-SHA384:TLS_AES_256_GCM_SHA384' -ciphersuites 'ECDHE-RSA-AES256-GCM-SHA384:TLS_AES_256_GCM_SHA384' xxxxx.xxxx.xx:443
Error with command: "-ciphersuites ECDHE-RSA-AES256-GCM-SHA384:TLS_AES_256_GCM_SHA384"
140272218875200:error:1426E0B9:SSL routines:ciphersuite_cb:no cipher match:../ssl/ssl_ciph.c:1294:
$ 
 ```
I stil think this "combination" needs to be explained in more detail as you did here.

**Comment by mattcaswell** [comment]
I also just thought they are also combined in the return value from functions such as `SSL_get_ciphers()`, `SSL_get_client_ciphers() and `SSL_get1_supported_ciphers()`, i.e. you set them separately - but when you get them back out again they are combined.

**Comment by mattcaswell** [comment]
> I stil think this "combination" needs to be explained in more detail - as you did here.

Fair comment

**Comment by DimitriPapadopoulos** [comment]
To further explain how I got confused, please refer to your comment https://github.com/openssl/openssl/issues/10936#issuecomment-578060278 a few months ago:
> No, this is working as designed. The ciphersuites you are seeing are TLSv1.3 ciphersuites.
> 
> TLSv1.2 (and below) ciphersuites are handled separately to TLSv1.3 ciphersuites. The two are incompatible with each other (you can't use a TLSv1.2 ciphersuite in TLSv1.3 and vice versa).

Of course the above refers to TLSv1.2 `-ciphers` vs. TLSv1.3 `-ciphersuites` options, not to the protocol level. No problem here. This is just to try to explain how it can get confusing for end-users who don't understand all the subtleties: _incompatible_ (in options) vs. _combined_ (at the protocol level).

**Comment by nhorman** [comment]
Marking as inactive, to be closed at the end of 3.4  dev, barring further input

**Comment by t8m** [comment]
Closing as the issue was labeled as inactive for extended period of time and there was no additional input.
Please add a comment if you have any further information to support reopening the issue.
